### PR TITLE
examples of how to handle properties

### PR DESCRIPTION
**Pull request by thedavidmeister**
### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Comment by timotree3** [comment]
This is great documentation but do you expect people to see this? Will this be linked to from somewhere?

**Comment by neonphog** [comment]
> This is great documentation but do you expect people to see this? Will this be linked to from somewhere?

api docs are where it's at, if folks aren't looking there, we need to train them to : )

File: crates/test_utils/wasm/wasm_workspace/zome_info/src/lib.rs [hunk 1]
```diff
@@ -72,6 +73,102 @@ fn dna_info(_: ()) -> ExternResult<DnaInfo> {
...
+struct PropertiesValue(Value);
+
+#[hdk_extern]
+fn dna_info_value(k: String) -> ExternResult<serde_yaml::Value> {
+    Ok(
+        PropertiesValue::try_from(hdk::prelude::dna_info()?.properties)?.0[k].clone()
+    )
+}
+
+/// Yaml doesnt enforce the type of any value.
```
Comment (neonphog) [inline crates/test_utils/wasm/wasm_workspace/zome_info/src/lib.rs:121]:
```suggestion
/// Yaml doesn't enforce the type of any value.
```

File: crates/test_utils/wasm/wasm_workspace/zome_info/src/lib.rs [hunk 2]
```diff
@@ -72,6 +73,102 @@ fn dna_info(_: ()) -> ExternResult<DnaInfo> {
...
+    foo: Option<Foo>,
+    /// This property can be absent but MUST be a `String` when present.
+    #[serde(default)]
+    bar: Option<String>,
+    /// The direct struct approach does not prevent us from using `Value` in
+    /// nested positions. `Value::Null` will be the default when the value is
+    /// not present in the properties.
+    #[serde(default)]
+    baz: Value,
+}
```
Comment (neonphog) [inline crates/test_utils/wasm/wasm_workspace/zome_info/src/lib.rs:151]:
```suggestion
}

```

**Review by neonphog — APPROVED** [review]
:+1:

**Comment by timotree3** [comment]
@neonphog what do you mean? It looks to me like all the documentation is inside a test DNA's source code. It wouldn't show up in any HDK docs

File: crates/test_utils/wasm/wasm_workspace/zome_info/src/lib.rs [hunk 3]
```diff
@@ -72,6 +73,103 @@ fn dna_info(_: ()) -> ExternResult<DnaInfo> {
...
+///
+/// - Loss of a declarative/typed structure that can be inspected visually/IDE
+/// - Inclusion of an additional dependency on `serde_yaml` in the WASM
+///
+/// The other option is to use `rmpv::Value` from the `rmpv` crate, but many
+/// types supported by messagepack are not supported by yaml anyway. Also the
+/// traversal support for moving through yaml mappings and using floats for
+/// keys is relatively poor in `rmpv` compared to `serde_yaml`.
+#[derive(Deserialize, Serialize, Debug, SerializedBytes)]
+struct PropertiesValue(Value);
```
Comment (freesig) [inline crates/test_utils/wasm/wasm_workspace/zome_info/src/lib.rs:112]: Why not just move the `YamlProperties` type to the `holochain_zome_types`?

**Review by freesig — CHANGES_REQUESTED** [review]

**Review by freesig — APPROVED** [review]

**Comment by freesig** [comment]
@thedavidmeister This PR is missing a changelog entry

**Pull request merged**